tst/functional/TestKeepsGoingUp.kt

```diff
@@ -147,34 +179,207 @@ class TeleportingPlayer(override val actor: Actor): Player {
     }
 }
 
-class BFSPlayer(override val actor: Actor): Player {
+/*class ExploratoryPlayerWithoutEndNode(override val actor: Actor) : Player {
     override var feedback: InvalidMoveException? = null
-    private lateinit var previousNodeCoordinate: Coordinate
-    fun setStartNodeCoordinate(startNodeCoordinate: Coordinate) {
-        println("x: "  + startNodeCoordinate.x + " y: " + startNodeCoordinate.y)
-        actor.addNextNodeCoordinate(coordinate = startNodeCoordinate)
-        previousNodeCoordinate = startNodeCoordinate
+
+    override fun chooseNextMove(): Action {
+
+    }*/
+
+class ExploratoryPlayerWithEndNode(override val actor: Actor, private val memory: Memory,
+                                   private val end: Coordinate): Player {
```

Comment:
hard to read unless every parameter is on its own line

Reply:
fixed this